src/main/java/ReadingRangeCount.java

```diff
@@ -0,0 +1,33 @@
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class ReadingRangeCount {
+    int []  data = {3,3,5,4,10,11,12};
```

Comment:
data is hardcoded to be part of the class ReadingRangeCount.
shouldn't data be an external input? Also testing with various input ranges is not possible in this case!
Please correct.